### PR TITLE
fix: email export of data

### LINKED ISSUE #16: fix: email export of data
On some systems, clicking the **Export saved data as JSON** button does not do anything.

https://user-images.githubusercontent.com/45038919/148683026-3072104f-77bd-4846-8e32-672626c1b4ab.mp4

For now, this only happens on my phone, @sshazril or @akabla could not reproduce, so we might drop this later if it only affects me, but it could be beneficial to keep track of it/document it if it ends up happening to someone else in the future.

### PR DESCRIPTION
**Pull request by adam-sroka**
This PR aims to fix #16, the email export of data button not working on some phones.

### CODE REVIEW

**Comment by adam-sroka** [comment]
@sshazril I just tested it with the new package, but it still doesn't work. We can leave this as it was before for now, and if no one else experiences this, then we could close it, as in the future we will anyways implement a new way of exporting data.

**Comment by adam-sroka** [comment]
As mentioned in PR #21:

Upon updating the Flutter SDK, the following warnings pop up when building for Android:
image
```
Note: /usr/local/Caskroom/flutter/2.5.0/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider_android-2.0.11/android/src/main/java/io/flutter/plugins/pathprovider/PathProviderPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: /usr/local/Caskroom/flutter/2.5.0/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_mailer-2.0.1/android/src/main/java/com/dataxad/flutter_mailer/FlutterMailerPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```

Since these are related to `path_provider` and `flutter_mailer`, I suggest this PR should also involve investigating moving away from them, into something like [share_plus](https://pub.dev/packages/share_plus).

**Comment by adam-sroka** [comment]
@sshazril let me know if this works for you — it does for me (finally!) and I can export the data to my email app or any other app as well.

**Review by sshazril — CHANGES_REQUESTED** [review]
Screenshot_20220217-123932_Gmail
Looks like something breaks somewhere. It works for me mostly and sometimes it does not.
I haven't figured out exactly how to reproduce the bug but during my testing I manage to get the error at least 4 times. The bug seems to be related to making changes to collection and then saving.

**Comment by adam-sroka** [comment]
> Screenshot_20220217-123932_Gmail Looks like something breaks somewhere. It works for me mostly and sometimes it does not. I haven't figured out exactly how to reproduce the bug but during my testing I manage to get the error at least 4 times. The bug seems to be related to making changes to collection and then saving.

Good catch, I'll try to reproduce and look into it.

File: lib/home/widgets/sync_screen.dart [hunk 1]
```diff
@@ -36,18 +36,7 @@ class SyncScreen extends StatelessWidget {
...
-                  subject: 'Gibsonify collection and recipe data',
-                  recipients: [],
-                  isHTML: true,
-                  bccRecipients: [],
-                  ccRecipients: [],
-                  attachments: [_collectionfilePath, _recipefilePath],
-                );
-
-                await FlutterMailer.send(mailOptions);
+                await Share.shareFiles([_recipefilePath, _collectionfilePath]);
```
[the remark this answers is not in the document]
Comment (adam-sroka) [reply lib/home/widgets/sync_screen.dart:39]: You're right, just implemented it 🙌.

**Comment by adam-sroka** [comment]
Hey @sshazril, I couldn't reproduce the bug — so I added a button to save to a local file as well. Hopefully this will work and we can merge this now — if the testing reveals problems, we can reopen this in the future. Let me know what you think 😊.

File: lib/home/widgets/sync_screen.dart [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:gibsonify/recipe/recipe.dart';
 import 'package:gibsonify/home/home.dart';
 import 'dart:convert';
 
 class SyncScreen extends StatelessWidget {
+  final _exportSubject = 'Gibsonify collection and recipe data';
+  final _exportText = 'Gibsonify collection and recipe data attached as a JSON '
+      'string. <br> Data can be pasted into '
```
Comment (sshazril) [inline lib/home/widgets/sync_screen.dart:13]: I don't think the share plugin has HTML support so can just drop the `<br>`

**Review by sshazril — APPROVED** [review]
Minor tweak but otherwise all good! Like the download button :)

**Pull request merged**